### PR TITLE
ISSUE-2711 yaml lint

### LINKED ISSUE #2711: [CI] Introduce yaml lint to ci and make lint
**Summary**
Currently we have no linter on yaml related workflow configuration, which introduced a lot of unneeded debugging time when new coming workflow went wrong, here by we could introduce yaml lint tool on run lint on CI and on local
Ref
https://github.com/adrienverge/yamllint
https://github.com/ibiqlik/action-yamllint

### PR DESCRIPTION
**Pull request by pinylin**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Introduce yaml lint tool to check syntax validity of workflow yaml files 

## Changelog

- Build/Testing/CI
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #2711 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: .github/workflows/yaml-lint.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (PsiACE) [inline .github/workflows/yaml-lint.yml:1]: rm this line

File: .github/workflows/yaml-lint.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+name: Yaml Lint
+on: [push]
```
Comment (PsiACE) [inline .github/workflows/yaml-lint.yml:3]: run only when changing yaml files

File: .yamllint.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+extends: default
```
Comment (PsiACE) [inline .yamllint.yml:1]: add this file to `.licenserc.yaml `

File: .github/workflows/yaml-lint.yml (hunk 2 above)
Comment (PsiACE) [reply .github/workflows/yaml-lint.yml:3]: also, push -> pull_request

File: .yamllint.yml (hunk 3 above)
Comment (pinylin) [reply .yamllint.yml:1]: ok

File: .github/workflows/yaml-lint.yml [hunk 4]
```diff
@@ -1,6 +1,9 @@
----
 name: Yaml Lint
-on: [push]
+on:
+  pull_request:
+    paths:
+      - '.github/**.yaml'
+      - '.github/**.yml'
```
Comment (PsiACE) [inline .github/workflows/yaml-lint.yml:6]:
```suggestion
      - '**/*.yaml'
      - '**/*.yml'
```

I guess maybe that would be better? We also need lint `.licenserc.yaml` etc.

Comment (PsiACE) [reply .github/workflows/yaml-lint.yml:6]: Can you verify this?

Comment (pinylin) [reply .github/workflows/yaml-lint.yml:6]: ok

**Review by PsiACE — COMMENTED** [review]
LGTM

**Review by PsiACE — APPROVED** [review]

**Comment by BohuTANG** [comment]
/lgtm
Good job, thank you @pinylin !

**Pull request merged**